src/yukkuri_game/game/systems/gossip_system.py

```diff
@@ -47,50 +51,91 @@ def on_social_interaction(self, event: SocialInteractionEvent) -> None:
             self._exchange_gossip(world, event.initiator_id, event.target_id)
             self._exchange_gossip(world, event.target_id, event.initiator_id)
 
-        # Handle Witnessing (Visual/Auditory)
-        # Use Pymunk Point Query
-        if not self.physics_system:
-             self.physics_system = world.services.try_get(PhysicsSystem)
+        # Handle Witnessing (Sector-based)
+        if not self.sector_map:
+            self.sector_map = world.services.try_get(SectorMap)
 
-        if self.physics_system and self.physics_system.space:
-             self._process_witnesses_spatial(world, event, actor_trans, now)
-        else:
-             # Fallback if physics system not ready (shouldn't happen in normal gameplay)
-             pass
+        if self.sector_map:
+             self._process_witnesses_sector(world, event, actor_trans, now)
 
-    def _process_witnesses_spatial(self, world: World, event: SocialInteractionEvent, actor_trans: Transform, now: float):
+    def _process_witnesses_sector(self, world: World, event: SocialInteractionEvent, actor_trans: Transform, now: float):
         """
-        Uses spatial query to find witnesses.
+        Uses SectorMap to find witnesses.
         """
-        visual_range = 300.0
-
-        # Pymunk query: query for shapes within visual_range of the actor
-        point = (actor_trans.x, actor_trans.y)
-        results = self.physics_system.space.point_query(point, visual_range, pymunk.ShapeFilter())
-
-        for shape_query_info in results:
-            shape = shape_query_info.shape
-            body = shape.body
-            if not body or not body.userdata:
-                continue
+        range_type = "visual"
+        if event.interaction_type in ["Scream", "Shout"]:
```

Comment:
Hard coding this will make it a pain when adding new interactions, make it more dynamic and data driven